pkg/plugin/instance.go

```diff
@@ -28,15 +29,20 @@ func NewGitHubInstance(ctx context.Context, settings models.Settings) (instancem
 }
 
 // NewDataSourceInstance creates a new instance
-func NewDataSourceInstance(_ context.Context, settings backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
-	datasourceSettings, err := models.LoadSettings(settings)
+func NewDataSourceInstance(ctx context.Context, settings backend.DataSourceInstanceSettings) (instancemgmt.Instance, error) {
+	opts, err := settings.HTTPClientOptions(ctx)
 	if err != nil {
 		return nil, err
 	}
 
+	datasourceSettings, err := models.LoadSettings(settings)
+	if err != nil {
+		return nil, err
+	}
+	
 	datasourceSettings.CachingEnabled = true
 
-	instance, err := NewGitHubInstance(context.Background(), datasourceSettings)
+	instance, err := NewGitHubInstance(context.Background(), datasourceSettings, opts)
```

Comment:
Do you know perhaps why we don't pass down the context from the `NewDataSourceInstance`?

Reply:
Let's do this then.